rice-middleware/kim/kim-impl/src/main/java/org/kuali/rice/kim/impl/group/GroupServiceImpl.java

```diff
@@ -91,6 +91,30 @@ public Group getGroup(String groupId) throws RiceIllegalArgumentException {
 		return Collections.unmodifiableList(new ArrayList<Group>( groups ));
     }
 
+    public List<Group> getParentGroupsByGroupsIds(List<String> groupIds, Set<Group> existingGroups) {
```

Comment:
Wouldn't this be a private or protected method with a helper which provides the initial value for existing groups that is public?